elide-graphql/src/main/java/com/yahoo/elide/graphql/federation/EntitiesDataFetcher.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2023, the original author or authors.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.graphql.federation;
+
+import com.yahoo.elide.core.PersistentResource;
+import com.yahoo.elide.core.exceptions.BadRequestException;
+import com.yahoo.elide.core.request.EntityProjection;
+import com.yahoo.elide.graphql.GraphQLRequestScope;
+import com.yahoo.elide.graphql.KeyWord;
+import com.yahoo.elide.graphql.containers.NodeContainer;
+
+import com.apollographql.federation.graphqljava._Entity;
+
+import org.apache.commons.lang3.StringUtils;
+
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import io.reactivex.Observable;
+
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Entities Data Fetcher for Apollo Federation.
+ */
+public class EntitiesDataFetcher implements DataFetcher<List<NodeContainer>> {
+
+    @Override
+    public List<NodeContainer> get(DataFetchingEnvironment environment) throws Exception {
+        List<Map<String, Object>> representations = environment.getArgument(_Entity.argumentName);
+        List<String> ids = representations.stream() // only supports the single id key field
+            .map(representation -> {
+                    String idKey = representation.keySet()
+                            .stream()
+                            .filter(key -> !KeyWord.TYPENAME.getName().equals(key))
+                            .findFirst()
+                            .get();
+                    return (String) representation.get(idKey);
+                })
+            .toList();
+
+        String entityName = StringUtils.uncapitalize(representations.get(0).get(KeyWord.TYPENAME.getName()).toString());
+
+        GraphQLRequestScope requestScope = environment.getLocalContext();
+        EntityProjection projection = requestScope
+                .getProjectionInfo()
+                .getProjection(null, entityName);
```

Comment:
What if the same entity is requested multiple times with different aliases or with an alias in general?

Reply:
The alias isn't part of the representations argument. If different aliases are used in the same query it is called for as many times.

For instance for this query on the supergraph

```
query {
  a:groupReviews {
    id
    stars
    text
    group {
      name
      commonName
    }
  }
  b:groupReviews {
    id
    stars
    text
    group {
      name
      commonName
    }
  }
}
```

The following query gets run twice on the subgraph

```json
{
  "query": "query($representations:[_Any!]!){_entities(representations:$representations){...on Group{commonName}}}",
  "variables": {
    "representations": [
      {
        "__typename": "Group",
        "name": "com.yahoo.elide"
      },
      {
        "__typename": "Group",
        "name": "com.example.repository"
      },
      {
        "__typename": "Group",
        "name": "com.test"
      }
    ]
  }
}
```